tests/test_get_avatar.py

```diff
@@ -0,0 +1,105 @@
+import unittest
+import json
+import requests as r
+import yaml
+import pytest
+import logging
+import username_api
+import os.path
+
+data = yaml.load(open(os.path.join('tests', 'test_data.yml')))
+websites = yaml.load(open('websites.yml'))
+
+class TestGet_avatar(object):
+
+	@pytest.mark.parametrize('user', data['github']['taken_usernames'])
```

Comment:
Can we have better test parameterisation in this PR? (specifically for the newly added tests) There is lot of duplicated logic already in tests.

Reply:
Okay, I will look into parameterisation. I would assume it's okay to use the same methods/parameterisation in the username_check tests. Is that fine?(Of course, I'll do those in a separate PR later)

Reply:
If more parameterisation is not easy, it can be deferred to GCI task https://github.com/manu-chroma/username-availability-checker/issues/36

Reply:
I'm going to shift us from pytest to nosetests as part of #36 I think. It has a pretty powerful library [parameterized](https://github.com/wolever/parameterized), which although supports py.test, isn't as well supported(not as pretty and I see quite a few bug reports) (see [FAQ](https://github.com/wolever/parameterized#faq) for more details).

This is probably a large change(to builds etc.) , so I'd want it to be in it's own PR, but let me know whatever you decide.

Reply:
nosetests is dead, and nose2 is stillborn.
It works well in pytest.  The caveat about the generated method names is of little consequence as usually test method names are hidden, and `name_func` can be used to create new names that are meaningful.

Reply:
Are we okay dealing [parameterized/#34](https://github.com/wolever/parameterized/issues/34) then?
Or can we just shift to unittest(since we probably won't need pytest with that)?

Reply:
Not our problem.  pytest devs are not going to delete remove support for that until their main plugins are fixed.  That is just an early warning system, and the alarm has been heard correctly.